### ISSUE TITLE
[BUG] Tuplex fails to decode the type when calling .unique() for 311 benchmark.

### ISSUE THREAD

**Issue by kchasialis** [tuplex/tuplex#99]
I downloaded a 311 requests dataset which contains 6 columns (one of them is "Incident Zip") in order to run the 311 benchmark (tuplex/benchmarks/311/runtuplex.py)

When the following code is executed:

```
tstart = time.time()
df = ctx.csv(
    ",".join(perf_paths),
    null_values=["Unspecified", "NO CLUE", "NA", "N/A", "0", ""],
    type_hints={0: typing.Optional[str],
                1: typing.Optional[str],
                2: typing.Optional[str],
                3: typing.Optional[str],
                4: typing.Optional[str],
                5: typing.Optional[str],
    },
)
# Do the pipeline
df = df.mapColumn("IncidentZip", fix_zip_codes).unique()
# Output to csv
df.tocsv(output_path)
job_time = time.time() - tstart
print(json.dumps({"startupTime": startup_time, "jobTime": job_time}))
```

I receive the following error:
[2022-04-26 16:57:10.185] [global] [error] decoding of other types not yet supported...

Initially I suspected that the issue might be that Incident Zip column is inferred as f64 so I explicitly set the type of all columns to be str, as seen in the code snippet above.

However, the error still persists. Any ideas?

PS: I believe that error is on unique() method because when I remove the call it works fine.

Thanks in advance.

**Comment by LeonhardFS** [comment]
Do you have by a chance a snippet of the input data you're using?

**Comment by LeonhardFS** [comment]
this could potentially be related to #96

**Comment by kchasialis** [comment]
head 311-service-requests-pruned-clean.csv

UniqueKey,CreatedDate,Agency,ComplaintType,Descriptor,IncidentZip,StreetName
46688741,06/30/2020 07:24:41 PM,NYPD,Noise - Residential,Loud Music/Party,10037.0,MADISON AVENUE
53493739,02/28/2022 07:30:31 PM,NYPD,Illegal Parking,Double Parked Blocking Traffic,11203.0,EAST   56 STREET
48262955,11/27/2020 12:00:00 PM,DSNY,Derelict Vehicles,Derelict Vehicles,11203.0,CLARKSON AVENUE
48262956,11/27/2020 12:00:00 PM,DSNY,Derelict Vehicles,Derelict Vehicles,11208.0,SHEPHERD AVENUE
48262957,11/27/2020 12:00:00 PM,DSNY,Derelict Vehicles,Derelict Vehicles,11238.0,BERGEN STREET
46688747,06/30/2020 02:51:45 PM,NYPD,Noise - Vehicle,Engine Idling,10009.0,EAST   12 STREET
46688748,06/30/2020 09:26:45 AM,NYPD,Non-Emergency Police Matter,Face Covering Violation,11204.0,20 AVENUE
48262973,11/27/2020 03:46:00 PM,DEP,Water Quality,unknown odor/taste in drinking water (QA6),10021.0,EAST   70 STREET
53493766,02/28/2022 05:28:38 AM,NYPD,Noise - Vehicle,Car/Truck Horn,11366.0,PARSONS BOULEVARD

I have also attached the Python code that I used.

**Comment by LeonhardFS** [comment]
Thanks a lot! I am still investigating, but pushed a preliminary fix on #112.
